### PR TITLE
Added text input

### PR DESCRIPTION
**Pull request by VictorKoenders**
Follow-up of #35

### CODE REVIEW

**Comment by 17cupsofcoffee** [comment]
This is really nice 😄 I might leave a couple of comments inline, but this is pretty much ready to go I think.

File: src/input.rs [hunk 1]
```diff
@@ -27,9 +28,21 @@ impl InputContext {
...
+    pub(crate) fn cleanup_after_state_update(&mut self) {
+        self.previous_key_state = self.current_key_state.clone();
+        self.current_text_input = None;
+    }
+}
+
+/// Returns the text that the user entered this tick.
+/// This will match the user's keyboard and OS settings.
+pub fn get_text_input(ctx: &Context) -> &Option<String> {
+    &ctx.input.current_text_input
```
Comment (17cupsofcoffee) [inline src/input.rs]: `Option<&str>` would be a little nicer of a type signature - you could get that by doing `ctx.input.current_text_input.as_ref().map(String::as_str)`.

File: examples/input.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            16.0,
+        );
+    }
+}
+
+impl State for GameState {
+    fn update(&mut self, ctx: &mut Context) {
+        if input::is_key_pressed(ctx, Key::Backspace) {
+            self.input.pop();
+            self.update_text();
```
Comment (17cupsofcoffee) [inline examples/input.rs]: There's a [`set_content`](https://docs.rs/tetra/0.1.6/tetra/graphics/text/struct.Text.html#method.set_content) method on `Text`, which will save you from having to recreate the object every time you update it.

File: src/input.rs (hunk 1 above)
Comment (VictorKoenders) [reply src/input.rs]: I actually had that the first time, but figured that it was messier than returning an `&Option<String>`

Comment (17cupsofcoffee) [reply src/input.rs]: A little messier on our end, but hopefully a little clearer in the docs :)

File: examples/input.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+use tetra::graphics::text::{Text, Font};
+use tetra::input::{self, Key};
+use tetra::{Context, ContextBuilder, State};
+
+struct GameState {
+    input: String,
+    text: Text,
+}
+
+impl GameState {
```
Comment (17cupsofcoffee) [inline examples/input.rs]: You can get rid of this impl block now, don't think it's being used anywhere else.

Comment (VictorKoenders) [reply examples/input.rs]: Derp, that's what I get for committing without testing

**Comment by 17cupsofcoffee** [comment]
Merged :D Thank you for the contribution!

**Pull request merged**